apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.spec.js

```diff
@@ -243,6 +243,80 @@ describe('notifyResidentsAboutNewsItem', () => {
         })
 
+        test('newsItem message is sent to the desired property and unitName', async () => {
+            const residentClient1 = await makeClientWithResidentUser()
+            const residentClient2 = await makeClientWithResidentUser()
+            const [o10n] = await createTestOrganization(adminClient)
+            const [property1] = await createTestProperty(adminClient, o10n)
+            const [property2] = await createTestProperty(adminClient, o10n)
+
+            const unitType1 = FLAT_UNIT_TYPE
+            const unitName1 = faker.lorem.word()
+
+            await createTestResident(adminClient, residentClient1.user, property1, {
+                unitType: unitType1,
+                unitName: unitName1,
+            })
+
+            // Second user to check the sending of messages with the same unitName and unitType
+            await createTestResident(adminClient, residentClient2.user, property2, {
+                unitType: unitType1,
+                unitName: unitName1,
+            })
+
+            const newsItemTitle = faker.lorem.words(100)
+            const newsItemBody = faker.lorem.words(100)
+
+            const [newsItem] = await createTestNewsItem(
+                adminClient,
+                o10n,
+                {
+                    title: newsItemTitle,
+                    body: newsItemBody,
+                }
+            )
+
+            await createTestNewsItemScope(adminClient, newsItem, {
+                property: { connect: { id: property1.id } },
+                unitType: unitType1,
+                unitName: unitName1,
+            })
+
+            const payload1 = getRandomTokenData({
+                devicePlatform: DEVICE_PLATFORM_ANDROID,
+                appId: APP_RESIDENT_ID_ANDROID,
+                pushToken: getRandomFakeSuccessToken(),
+            })
+
+            const payload2 = getRandomTokenData({
+                devicePlatform: DEVICE_PLATFORM_ANDROID,
+                appId: APP_RESIDENT_ID_ANDROID,
+                pushToken: getRandomFakeSuccessToken(),
+            })
+
+            await syncRemoteClientByTestClient(residentClient1, payload1)
+            await syncRemoteClientByTestClient(residentClient2, payload2)
+
+            const messageWhere1 = { user: { id: residentClient1.user.id }, type: NEWS_ITEM_COMMON_MESSAGE_TYPE }
+            const messageWhere2 = { user: { id: residentClient2.user.id }, type: NEWS_ITEM_COMMON_MESSAGE_TYPE }
+
+            // Publish NewsItem to make it sendable
+            await publishTestNewsItem(adminClient, newsItem.id)
+            await notifyResidentsAboutNewsItem(newsItem.id)
+
+            // The message should not be sent to the second user with a different address, but the same unitName and unitType 
+            await waitFor(async () => {
+                const messages1 = await Message.getAll(adminClient, messageWhere1)
+                const messages2 = await Message.getAll(adminClient, messageWhere2)
```

Comment:
you can use `.getOne` here and check `expect(message1).toBeDefined()` and `expect(message2).toBeUndefined()` below

Reply:
Okey, I am use getOne, but message2 receives [], which will not pass the test toBeUndefinded()

Reply:
Ow, method getAll return only object, not array. Okey, I will learn it, thanks^